File: factorial.py
def factorial(num:int) -> int:
  fac = 1
  while num > 0:
    fac *= num
    num-=1
  return fac

def factorial_r(num:int) -> int:
  if num > 0:
    return num * factorial(num-1)
  elif num == 0 or num == 1:
    return 1
  else:
    return None

print(factorial(5))
print(factorial_r(5))
print(factorial_r(1))
print(factorial_r(0))
print(factorial_r(-1))

File: factorial.cpp
#include <iostream>

unsigned long factorial(unsigned long num) {
  unsigned long fac = 1;
  while(num > 0) {
    fac *= num;
    num--;
  }
  return fac;
}

unsigned long factorial_r(unsigned long num) {

  if(num > 0) {
    return num*factorial_r(num-1);
  }
  else
    return 1;
}

int main() {

  std::cout << factorial(5) << std::endl;
  std::cout << factorial_r(5) << std::endl;  
}
